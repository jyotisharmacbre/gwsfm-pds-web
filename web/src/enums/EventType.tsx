enum EventType {
  save,
  next,
  previous,
  none
}

export default EventType;
