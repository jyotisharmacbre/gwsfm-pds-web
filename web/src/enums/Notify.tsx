enum Notify {
  none,
  success,
  error
}

export default Notify;
