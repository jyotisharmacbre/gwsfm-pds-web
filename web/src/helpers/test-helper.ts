import checkPropTypes from 'check-prop-types';

export const findByTestAtrr = (component: any, attr: string) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component: any, expectedProps: any) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    'props',
    component.name
  );
  return propsErr;
};
