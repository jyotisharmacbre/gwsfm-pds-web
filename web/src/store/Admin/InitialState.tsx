import { IAdminState } from './Types/IAdminState';
import { IAdminDefaults } from './Types/IAdminDefault';

export const initialState: IAdminState = {
	adminDefaultValues: [],
	error: ''
};
