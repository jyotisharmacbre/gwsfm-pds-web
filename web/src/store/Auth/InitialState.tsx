import { IAuth } from "./Types/IAuth";

export const initialState: IAuth= {
    token: null,
    name: null,
    firstName: null,
    email: null
};
