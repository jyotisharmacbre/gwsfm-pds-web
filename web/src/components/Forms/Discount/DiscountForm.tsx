import React, { Component } from 'react';
import {
  Field,
  reduxForm,
  InjectedFormProps,
  formValueSelector
} from 'redux-form';
import PdsFormInput from '../../PdsFormHandlers/PdsFormInput';
import PdsFormSelect from '../../PdsFormHandlers/PdsFormSelect';
import PdsFormTextArea from '../../PdsFormHandlers/PdsFormTextArea';
import PdsFormButton from '../../PdsFormHandlers/PdsFormButton';
import { selectionButtons } from '../../../helpers/constants';
import { Validate, alphaNumeric, onlyNumber } from '../../../helpers/fieldValidations';
import { connect } from 'react-redux';
import { IState } from '../../../store/state';
import { FormattedMessage, injectIntl } from 'react-intl';
import { LookupType } from '../../../store/Lookups/Types/LookupType';
import { getDropdown } from '../../../helpers/utility-helper';
import PdsFormTypeAhead from '../../PdsFormHandlers/PdsFormTypeAhead';
import { IProjectDetail } from '../../../store/CustomerEnquiryForm/Types/IProjectDetail';
import { ICurrency } from '../../../store/Lookups/Types/ICurrency';
import IReactIntl from '../../../Translations/IReactIntl';

interface Props{

}

const DiscountForm: React.FC<
  Props & IReactIntl & InjectedFormProps<any>
> = (props: any) => {
  
    return (
      <div className="row">
        <div className="col-lg-12">
          <div className="Discountforms_wrap">
            <div className="row">
              <div className="col-lg-7">
                <form className="custom-wrap p-0">
                  <h2><FormattedMessage id="TITLE_SUBCONTRACTOR_DISCOUNT"></FormattedMessage></h2>
                    <Field
                      name="name"
                      type="text"
                      component={PdsFormInput}
                      className="required"
                      validate={[
                        Validate.required('LABEL_SUPPLIER'),
                        Validate.maxLength(1000)
                      ]}
                      messageKey="MESSAGE_SUPPLIER_NAME"
                      labelKey="LABEL_SUPPLIER"
                      placeholderKey="PLACEHOLDER_ENTER_SUPPLIER_NAME"
                    />
                    <Field
                      name="name"
                      type="text"
                      component={PdsFormInput}
                      className="required"
                      validate={[
                        Validate.required('LABEL_STATE_DETAILS'),
                        Validate.maxLength(1000)
                      ]}
                      messageKey="MESSAGE_STATE_DETAILS_NAME"
                      labelKey="LABEL_STATE_DETAILS"
                      placeholderKey="PLACEHOLDER_ENTER_STATE_DETAILS"
                    /> 
                  <div className="form-group">
                    <label><FormattedMessage id="LABEL_TOTAL_DISCOUNT"></FormattedMessage></label>
                    <input
                      className="width-120 form-control"
                      type="text"
                      placeholder=""
                    />
                    <span className="symbol_fix">&#163;</span>
                    <label className="w-100 mb-0 mt-3">
                    <FormattedMessage id="LABEL_SUB_TOTAL_DISCOUNTS"></FormattedMessage> 
                    </label>
                    <label className="m-0"><FormattedMessage id="LABEL_200000"></FormattedMessage></label>
                    </div>
                  <Field
                  name="scope"
                  rows="6"
                  component={PdsFormTextArea}
                  className="required"
                  validate={[
                    Validate.required('LABEL_COMMENTS'),
                    Validate.maxLength(1040)
                  ]}             
                  labelKey="LABEL_COMMENTS"
                  placeholderKey="PLACEHOLDER_TYPE_IN_ANY_ADDITIONAL_COMMENTS"
                />
                </form>
              </div>
            </div>

            <div className="hr_line"></div>

            <div className="row">
              <div className="col-lg-7">
                <form className="custom-wrap p-0">
                  <h2><FormattedMessage id="TITAL_CLIENT_DISCOUNT"></FormattedMessage></h2>
                  <Field
                      name="name"
                      type="text"
                      component={PdsFormInput}
                      className="required"
                      validate={[
                        Validate.required('LABEL_CLIENT'),
                        Validate.maxLength(1000)
                      ]}
                      messageKey="MESSAGE_CLIENT_NAME"
                      labelKey="LABEL_CLIENT"
                      placeholderKey="PLACEHOLDER_ENTER_CLIENT_NAME"
                    />
                  <Field
                      name="name"
                      type="text"
                      component={PdsFormInput}
                      className="required"
                      validate={[
                        Validate.required('STATE_DETAILS'),
                        Validate.maxLength(1000)
                      ]}
                      messageKey="MESSAGE_STATE_DETAILS"
                      labelKey="LABEL_STATE_DETAILS"
                      placeholderKey="PLACEHOLDER_ENTER_STATE_DETAILS"
                    />
                  <div className="form-group">
                    <label><FormattedMessage id="LABEL_DISCOUNT_TYPE"></FormattedMessage></label>
                    <div className="form-check">
                      <input
                        type="radio"
                        className="form-check-input"
                        name="Value"
                        id="1"
                      />
                      <label className="form-check-label" htmlFor="1">
                      <FormattedMessage id="LABEL_PERCENTAGE"></FormattedMessage>
                      </label>
                    </div>
                    <div className="form-check">
                      <input
                        type="radio"
                        className="form-check-input"
                        name="Value"
                        id="2"
                      />
                      <label className="form-check-label" htmlFor="2">
                        LABEL_VALUE
                      </label>
                    </div>
                  </div>
                  <div className="form-group">
                    <label>LABEL_DISCOUNT</label>
                    <input
                      className="width-250 form-control"
                      type="text"
                      placeholder=""
                    />
                    <span className="symbol_fix">PLACEHOLDER_QUOTE_SUPPLIER2</span>
                    <label className="w-100 mb-0 mt-3">
                      LABEL_SUB_TOTAL_DISCOUNTS
                    </label>
                    <label className="m-0">LABEL_&#163;2,000.00</label>
                  </div>
                  <Field
                  name="scope"
                  rows="6"
                  component={PdsFormTextArea}
                  className="required"
                  validate={[
                    Validate.required('LABEL_COMMENTS'),
                    Validate.maxLength(1040)
                  ]}             
                  labelKey="LABEL_COMMENTS"
                  placeholderKey="PLACEHOLDER_TYPE_IN_ANY_ADDITIONAL_COMMENTS"
                />
                </form>
              </div>
            </div> 

            <div className="hr_line mb-0 mt-4"></div>

            <div className="mr-35 three-btn">
              <button className="active" type="button">
                BUTTON_PREVIOUS
              </button>
              <button type="button" name="next" className="active ml-auto">
                BUTTON_SAVE
              </button>
              <button type="button" name="next">
                BUTTON_NEXT
              </button>
            </div>
          </div>
        </div>
      </div>
    );
  }

  const form = reduxForm<IProjectDetail, Props>({
    form: 'DiscountForm',
    enableReinitialize: true
  })(injectIntl(DiscountForm));
  
  export default connect(null)(form);


