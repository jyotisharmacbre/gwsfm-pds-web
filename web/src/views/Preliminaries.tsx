import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { IState } from '../store/state';
import { IPreliminariesComponentDetails } from '../store/Preliminaries/Types/IPreliminariesComponentDetails';
import * as actions from '../store/rootActions';
import PreliminaryForm from '../components/Forms/PreliminaryForm/PreliminaryForm';
import { IPreliminaries } from '../store/Preliminaries/Types/IPreliminaries';
import { convertIntoDatabaseModel } from "../store/Preliminaries/DataWrapper";
import { toast } from 'react-toastify';
import Notify from '../enums/Notify';
import { ILookup } from '../store/Lookups/Types/ILookup';
import { ICurrency } from '../store/Lookups/Types/ICurrency';
import { FormattedMessage } from 'react-intl';
import EventType from '../enums/EventType';
import CalculationsSummaryTable from '../components/Table/CalculationsSummaryTable';
import CalculationsSummaryType from '../enums/CalculationsSummaryType';
import { getPropertyName,getFilterElementFromArray } from '../helpers/utility-helper';
import { formValueSelector } from 'redux-form';
import Currency from '../store/Lookups/InitialState/Currency';
import { History } from 'history';
interface IMapStateToProps {
  preliminaryDetails: Array<IPreliminariesComponentDetails>;
  lookupData: Array<ILookup>;
  currencies: Array<ICurrency> | null;
  match: any;
  notify: Notify;
  event: EventType;
  currencyId: number;
  preliminaryForm:Array<IPreliminariesComponentDetails>;
  history:History;
}
interface IMapDispatchToProps {
  preliminaryAdd: (
    preliminaryDetails: Array<IPreliminaries>,
    event: EventType
  ) => void;
  preliminaryEdit: (
    preliminaryDetails: Array<IPreliminaries>,
    event: EventType
  ) => void;
  getPreliminaryDetails: (projectId: string) => void;
  updateInputField: (inputData: any) => void;
  getAllCurrencies: () => void;
  getProjectStatus: () => void;
  getProjectDetail: (projectId: string) => void;
}

const Preliminaries: React.FC<
  IMapStateToProps & IMapDispatchToProps
> = props => {
  let paramProjectId: string = '';
  const CurrencyObj = new Currency();
  let isLookupSessionExists: boolean = (sessionStorage.getItem("lookupData") != null && sessionStorage.getItem("lookupData") != undefined && sessionStorage.getItem("lookupData") != "")
  useEffect(() => {
    window.scrollTo(0, 0);
    paramProjectId=props.match.params.projectId;
    props.getProjectDetail(paramProjectId);
    props.getProjectStatus();
    props.getAllCurrencies();
    if (paramProjectId != null && paramProjectId != '' && paramProjectId != undefined && isLookupSessionExists) {
      props.getPreliminaryDetails(paramProjectId);
    }
  }, []);
  useEffect(() => {
    window.scrollTo(0, 0);
    if (props.notify == Notify.success) {
      toast.success('Data Saved Successfully');
      if (props.event == EventType.next) {
        props.history.push('/Subcontractor/' + props.match.params.projectId);
      }
    }
    else if (props.notify == Notify.error) {
      toast.error('Error occured.Please contact to administrator.');
    }

  }, [props.notify, props.event]);
  useEffect(() => {
    if (paramProjectId != null && paramProjectId != '' && paramProjectId != undefined && isLookupSessionExists) {
      props.getPreliminaryDetails(paramProjectId);
    }
  }, [props.lookupData]);
  const handleExpandAllEvent = () => {
    var element: any = document.getElementsByClassName('expandAll');
    for (let i = 0; i < element.length; i++) {
      element[i].classList.remove('hide');
      element[i].classList.add('show');
    }
  };
  const handleToggle = (id: string) => {
    var element: any = document.getElementById('collapse_' + id);
    if (element != null) {
      var isClassExists = element.classList.contains('show');
      if (isClassExists) {
        element.classList.add('hide');
        element.classList.remove('show');
      } else {
        element.classList.remove('hide');
        element.classList.add('show');
      }
    }
  };
  const handleSaveData = (
    saveAll: boolean,
    event: EventType,
    preliminaryDetails: any,
    index: number

  ) => {
    var editData: Array<IPreliminaries> = [];
    var saveData: Array<IPreliminaries> = [];
    paramProjectId = props.match.params.projectId;
    let preData: Array<IPreliminariesComponentDetails> = [];
    preData.push(preliminaryDetails.preliminaryDetails[index]);
    var preliminariesData: Array<IPreliminaries> = saveAll ?
      convertIntoDatabaseModel(preliminaryDetails.preliminaryDetails, paramProjectId) :
      convertIntoDatabaseModel(preData, paramProjectId)
    editData = preliminariesData.filter((data) => {
      return (data.TotalCost > 0 && data.PreliminaryId != '');
    })
    saveData = preliminariesData.filter((data) => {
      return (data.TotalCost > 0 && data.PreliminaryId == '');
    })
    if (editData.length > 0 && saveData.length > 0) { props.preliminaryEdit(editData, event); props.preliminaryAdd(saveData, event); }
    else if (editData.length > 0) { props.preliminaryEdit(editData, event); }
    else if (saveData.length > 0) { props.preliminaryAdd(saveData, event) }
    else { toast.error('No data changed to save.'); }
  };
  const handlePrevious = () => {
    props.history.push(`/Project/${props.match.params.projectId}`);
  };

  return (
    <div className="container-fluid">
      <div className=" row">
        <div className="col-lg-12">
          <form className="custom-wrap">
            <div className="heading-subtitle">
              <h1>
                <span className="d-md-block d-none">
                  <FormattedMessage id='TITLE_JUSTIFICATION'></FormattedMessage>
                </span>
                <span className="d-md-none"> <FormattedMessage id='TITLE_JUSTIFICATION_SHORT'></FormattedMessage></span>
              </h1>
              <p className="text-green"> <FormattedMessage id='PAGE_PRELIMINARY_TITLE'></FormattedMessage></p>
            </div>

            <div className="row">
              <div className="col-lg-12">
                <div className="row">
                  <div className="col-lg-8">
                    <div className="table-responsive">
                      <CalculationsSummaryTable
                      projectId={props.match.params.projectId}
                      name={CalculationsSummaryType.preliminary} 
                      preliminary={props.preliminaryForm}
                      currencySymbol={getFilterElementFromArray(
                        props.currencies,
                        getPropertyName(
                        CurrencyObj,
                        prop => prop.currencyId
                      ),
                        props.currencyId,
                        getPropertyName(
                        CurrencyObj,
                        prop => prop.currencySymbol
                      )
                      )}/>
                    </div>
                  </div>
                  <div className="col-lg-4">
                    <div className="text-right">
                      <button type="button" className="active fltRght" onClick={() => handleExpandAllEvent()}>EXPAND ALL</button>
                    </div>
                  </div>
                </div>
              </div>
            </div>

            <div>
              <PreliminaryForm
                onSave={handleSaveData}
                onPrevious={handlePrevious}
                onToggle={handleToggle}
                preliminariesDetails={props.preliminaryDetails}
                currencySymbol={getFilterElementFromArray(
                    props.currencies,
                    getPropertyName(
                    CurrencyObj,
                    prop => prop.currencyId
                  ),
                    props.currencyId,
                    getPropertyName(
                    CurrencyObj,
                    prop => prop.currencySymbol
                  )
                  )}
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

const selector = formValueSelector('PreliminaryForm');


const mapStateToProps = (state: IState) => {
  return {
    preliminaryDetails: state.preliminary.preliminaryDetails,
    lookupData: state.lookup.projectstatus,
    currencies: state.lookup.currencies,
    notify: state.preliminary.notify,
    currencyId: state.project.form.currencyId,
    event: state.preliminary.event,
    preliminaryForm: selector(state, 'preliminaryDetails')
  };
};

const mapDispatchToProps = dispatch => {
  return {
    preliminaryAdd: (preliminaryDetails, event) =>
      dispatch(actions.preliminaryAdd(preliminaryDetails, event)),
    preliminaryEdit: (preliminaryDetails, event) =>
      dispatch(actions.preliminaryEdit(preliminaryDetails, event)),
    getProjectDetail: (projectId: string) =>
      dispatch(actions.getProjectDetail(projectId)),
    getPreliminaryDetails: (projectId: string) =>
      dispatch(actions.getPreliminaryDetails(projectId)),
    getAllCurrencies: () => dispatch(actions.getAllCurrencies()),
    getProjectStatus: () => dispatch(actions.getProjectStatus())
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Preliminaries);
