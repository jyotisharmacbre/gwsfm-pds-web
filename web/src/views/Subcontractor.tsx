import React,{useEffect} from 'react';
import { connect } from 'react-redux';
import SubcontractorForm from '../components/Forms/Subcontractor/SubcontractorForm';
import { ISubContractor } from '../store/SubContractor/Types/ISubContractor';
import * as actions from '../store/rootActions';
import { IState } from '../store/state';
import EventType from '../enums/EventType';
import { toast } from 'react-toastify';
import Notify from '../enums/Notify';
import {getPropertyName,getFilterElementFromArray} from '../helpers/utility-helper';
import { ICurrency } from '../store/Lookups/Types/ICurrency';
import { FormattedMessage } from 'react-intl';
import Currency from '../store/Lookups/InitialState/Currency';
import ProjectStatus from '../enums/ProjectStatus';
import {History} from "history";

interface IProps {
  match: any;
  history:History;
} 

interface IMapStateToProps {
  form:ISubContractor;
  notify: Notify;
  event: EventType;
  currencyId:number,
  currencies: Array<ICurrency> | null;
  status:number;
}

interface IMapDispatchToProps {
  subContractorFormAdd: (
    projectId: string,
    form: ISubContractor,
    event: EventType
  ) => void;
  subContractorFormEdit: (
    projectId:string,
    form: ISubContractor,
    event: EventType
  ) => void;
  getSubContractor: (projectId: string) => void;
  getProjectDetail: (projectId: string) => void;
  resetSubContractorState: () => void;
  getAllCurrencies:() => void;
}

const Subcontractor: React.FC<IProps & IMapStateToProps & IMapDispatchToProps> = props => {
  let paramProjectId:string = '';
  const CurrencyObj = new Currency();
  const [currencySymbol,setCurrencySymbol] = React.useState('$');
  
  useEffect(() => {
    window.scrollTo(0, 0);
    let allc = props.currencies;
    props.getAllCurrencies();
    paramProjectId = props.match.params.projectId;
    if (paramProjectId != null && paramProjectId != '') {
      props.getProjectDetail(paramProjectId);
      props.getSubContractor(paramProjectId);
    }
  }, []);

  useEffect(() => {
    if(props.currencyId>0 && props.currencies){
      setCurrencySymbol(
        getFilterElementFromArray(
                        props.currencies,
                        getPropertyName(
                        CurrencyObj,
                        prop => prop.currencyId
                      ),
                        props.currencyId,
                        getPropertyName(
                        CurrencyObj,
                        prop => prop.currencySymbol
                      )
                      )
      )
    }
  }, [props.currencyId,props.currencies]);
 
  useEffect(() => {
    if (props.notify == Notify.success) {
      if (props.event == EventType.next) {
        toast.success('Data Saved Successfully');
        props.history.push(`/Discounts/${props.match.params.projectId}`);
      } else if (props.event == EventType.previous) {
        toast.success('Data Saved Successfully');
        props.history.push(`/preliminaries/${props.match.params.projectId}`);
      }
      else if (props.event == EventType.save) {
        toast.success('Data Saved Successfully');
      }
      props.resetSubContractorState();
    }
  }, [props.notify, props.event]);

  const handleEvent= (data: ISubContractor,event:EventType) => {
    paramProjectId = props.match.params.projectId;
    data.activities[0].subContrActivityId == ''
      ? props.subContractorFormAdd(paramProjectId, data, event)
      : props.subContractorFormEdit(paramProjectId,data, event);
  }; 

  return (
    <div className="container-fluid">
           <div data-test="sub_row_status" className={(props.status==ProjectStatus.BidLost||props.status==ProjectStatus.OnHold)?"link_disabled row":"row"}>
          <div className="col-lg-12">
            <div className="custom-wrap">
              <div className="heading-subtitle">
                <h1>
                  <span className="d-md-block d-none">
                   <FormattedMessage id='TITLE_JUSTIFICATION'></FormattedMessage>
                  </span>
                  <span className="d-md-none"> <FormattedMessage id='TITLE_JUSTIFICATION_SHORT'></FormattedMessage></span>
                </h1>
                <p className="text-green"> <FormattedMessage id='PAGE_SUB_TITLE'></FormattedMessage></p>
              </div>
            {currencySymbol != '' ? <SubcontractorForm
              projectId={props.match.params.projectId}
              onSubmitForm={handleEvent}
              currencySymbol={currencySymbol}
            />:null}
          </div>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state: IState) => ({
  form: state.subContractor.form,
  notify: state.subContractor.notify,
  event: state.subContractor.event,
  currencyId:state.project.form.currencyId,
  currencies: state.lookup.currencies,
  status:state.project.form.status

});

const mapDispatchToProps = dispatch => {
  return {
    subContractorFormAdd: (projectId, form, event) =>
      dispatch(actions.subContractorFormAdd(projectId, form, event)),
    subContractorFormEdit: (projectId,form, event) =>
      dispatch(actions.subContractorFormEdit(projectId,form, event)),
    getProjectDetail: projectId =>
      dispatch(actions.getProjectDetail(projectId)),
    getSubContractor: projectId =>
      dispatch(actions.getSubContractor(projectId)),
    resetSubContractorState: () =>
      dispatch(actions.resetSubContractorState()),
    getAllCurrencies: () => dispatch(actions.getAllCurrencies())
    };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Subcontractor);