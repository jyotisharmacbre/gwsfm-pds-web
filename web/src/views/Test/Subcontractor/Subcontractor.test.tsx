import React from 'react';
import { mount } from 'enzyme';
import Subcontractor from '../../Subcontractor';
import {findByTestAtrr} from '../../../helpers/test-helper';
import { IntlProvider } from 'react-intl';
import translations from '../../../Translations/translation';
import EventType from '../../../enums/EventType';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { initialState } from '../../../store/Preliminaries/InitialState';
import { initialState as subcontractorInitialState} from '../../../store/SubContractor/InitialState';
import { lookUpInitialState, customerEnquiryInitialState } from '../Preliminaries/PreliminariesTestData';
import Notify from '../../../enums/Notify';
import {initialState as discountInitialState} from '../../../store/DiscountForm/InitialState';
import {initialState as summaryCalculationState} from '../../../store/SummaryCalculation/InitialState';
const mockStore = configureStore([]);
let store;
let wrapper;
const setUpStore=(initialState,lookUpInitialState,customerEnquiryInitialState,subcontractorInitialState)=>{
    store= mockStore({
    preliminary: initialState,
    lookup: lookUpInitialState,
    project: customerEnquiryInitialState,
    subContractor:subcontractorInitialState,
    discount:discountInitialState,
    summaryCalculation:summaryCalculationState,
});
store.dispatch=jest.fn();
}
const mountPreliminaryComponent=(Props)=>{
    wrapper = mount(
        <Provider store={store}>
        <IntlProvider locale="en" messages={translations['en'].messages}>
        <Subcontractor {...Props} />
        </IntlProvider>
        </Provider>
      );
    }
describe('Subcontractor component test cases', () => {
  const Props:any ={
    form:subcontractorInitialState.form,
    notify: Notify.none,
    event: EventType.none,
    currencyId:1,
    currencies:[] ,
    status:1,
    history:{ push: jest.fn() },
    match: {params: {projectId: 1}},
    subContractorFormAdd:jest.fn(),
    subContractorFormEdit:jest.fn(),
    getSubContractor:jest.fn(),
    getProjectDetail:jest.fn(),
    resetSubContractorState:jest.fn(),
    getAllCurrencies:jest.fn(),
    
  }
  beforeEach(() => {
    setUpStore(initialState,lookUpInitialState,customerEnquiryInitialState,subcontractorInitialState);
    mountPreliminaryComponent(Props);
  });
  
  it('defines the component', () => {
    expect(wrapper).toBeDefined();
  }); 
 
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should not make subcontractor element into readonly if project status is not bidlost or onhold', () => {
    let container=findByTestAtrr(wrapper,"sub_row_status").first();
    expect(container.hasClass("row")).toBe(true);
  });
  it('should make subcontractor element into readonly if project status is not bidlost or onhold', () => {
    customerEnquiryInitialState.form.status=4;
    setUpStore(initialState,lookUpInitialState,customerEnquiryInitialState,subcontractorInitialState);
    mountPreliminaryComponent(Props);
    let container=findByTestAtrr(wrapper,"sub_row_status").first();
    expect(container.hasClass("link_disabled")).toBe(true);
  });

}); 
