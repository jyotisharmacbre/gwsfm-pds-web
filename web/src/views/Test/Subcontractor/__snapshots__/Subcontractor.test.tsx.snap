// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subcontractor component test cases should match the snapshot 1`] = `ReactWrapper {}`;
